network/eni/eni_test.go

```diff
@@ -0,0 +1,33 @@
+package eni
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetInterfaceByMac(t *testing.T) {
+	mac, _ := net.ParseMAC("12:34:56:78:9a:bc")
+	interfaces := []net.Interface{
+		{
+			Index: 1,
+			Name: "eth1.1",
+			HardwareAddr: mac,
+		},
+		{
+			Index: 2,
+			Name: "eth1",
+			HardwareAddr: mac,
+		},
+		{
+			Index: 3,
+			Name: "eth1.1.1",
+			HardwareAddr: mac,
+		},
```

Comment:
Thanks for adding tests. For testing an interesting corner case, you can add one more interface here with a three letter name but a different MAC address. To make sure MAC address match is prioritized.

Reply:
updated the test